x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/Authentication.java

```diff
@@ -63,6 +63,14 @@
 import static org.elasticsearch.xpack.core.security.authc.AuthenticationField.FALLBACK_REALM_TYPE;
 import static org.elasticsearch.xpack.core.security.authc.RealmDomain.REALM_DOMAIN_PARSER;
 
+/**
+ * The Authentication class encapsulates identity information created after successful authentication
+ * and is the starting point of subsequent authorization.
+ *
+ * Authentication is serialized and travels across the cluster nodes as the sub-requests are handled,
+ * and can also be cached by long-running jobs that continue to act on behalf of the user, beyond
+ * the lifetime of the original request.
```

Comment:
Optional: could make sense to add a note along the lines of "Use getEffectiveSubject().getVersion() to retrieve the version of an authentication instance." to the Javadoc.

Reply:
It would be helpful to add some more comments in the javadoc for the Authenticaton to explain its current structure and how it should be reasoned about. I'll make it in a separate PR.